### PR TITLE
.Net: Sample demonstrating function advertisement depending on context

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
Closes: https://github.com/microsoft/semantic-kernel/issues/8840

### CODE REVIEW

File: dotnet/samples/Concepts/FunctionCalling/ContextDependentAdvertising.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+namespace FunctionCalling;
+
+/// <summary>
+/// These samples demonstrate how to advertise functions to AI model based on a context.
+/// </summary>
+public class ContextDependentAdvertising(ITestOutputHelper output) : BaseTest(output)
+{
+    /// <summary>
+    /// This sample demonstrates how to advertise functions to AI model based on the context of the chat history.
+    /// It advertise functions to the AI model based on the game sate.
```
Comment (dmytrostruk) [inline dotnet/samples/Concepts/FunctionCalling/ContextDependentAdvertising.cs:16]:
```suggestion
    /// It advertises functions to the AI model based on the game state.
```

File: dotnet/samples/Concepts/FunctionCalling/ContextDependentAdvertising.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            // Prompt the AI model.
+            ChatMessageContent result = await chatCompletionService.GetChatMessageContentAsync(chatHistory, settings, kernel);
+
+            Console.WriteLine(result);
+
+            iteration++;
+        }
+    }
+
+    private static Kernel CreateKernel()
```
Comment (RogerBarreto) [inline dotnet/samples/Concepts/FunctionCalling/ContextDependentAdvertising.cs:80]: if you are using this function just one, prefer keeping it in the main code.

```suggestion
```

File: dotnet/samples/Concepts/FunctionCalling/ContextDependentAdvertising.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        int iteration = 0;
+
+        // Define the functions for AI model to call.
+        var gameUtils = kernel.ImportPluginFromType<GameUtils>();
+        KernelFunction createMaze = gameUtils["CreateMaze"];
+        KernelFunction addTraps = gameUtils["AddTrapsToMaze"];
+        KernelFunction addTreasures = gameUtils["AddTreasuresToMaze"];
+        KernelFunction playGame = gameUtils["PlayGame"];
+
+        ChatHistory chatHistory = new();
```
Comment (RogerBarreto) [inline dotnet/samples/Concepts/FunctionCalling/ContextDependentAdvertising.cs:38]: nit.

If we added a implicit `string` operator to our `ChatMessageContent` that imply a `User` role with the `string` as `Content` property we could do similar to the `OpenAI SDK`, which is handy. `ChatHistory chatHistory = ["I would like to play..."]`

```suggestion
        ChatHistory chatHistory = [];
```

File: dotnet/samples/Concepts/FunctionCalling/ContextDependentAdvertising.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    {
+        // Create kernel
+        IKernelBuilder builder = Kernel.CreateBuilder();
+
+        builder.AddOpenAIChatCompletion(TestConfiguration.OpenAI.ChatModelId, TestConfiguration.OpenAI.ApiKey);
+
+        return builder.Build();
+    }
+
+    public sealed class GameUtils
```
Comment (dmytrostruk) [inline dotnet/samples/Concepts/FunctionCalling/ContextDependentAdvertising.cs:90]:
```suggestion
    private sealed class GameUtils
```

**Review by dmytrostruk — APPROVED** [review]

**Review by RogerBarreto — APPROVED** [review]

**Pull request merged**